src/components/AssignTask/AssignTask.js

```diff
@@ -1,63 +1,147 @@
+                    key={`${first_name}${last_name}${language.from_langauge}${language.to_langauge} language(s) ${y}`}
+                  >
+                    {lang_short[language.from_langauge]} &#9658;{" "}
+                    {lang_short[language.to_language]}
+                  </span>
+                ))}
+              </div>
             </div>
-
+            <p>
+              <b>Task Assignment</b>
+            </p>
+            <table className="uk-table uk-table-divider">
+              <thead>
+                <tr>
+                  <th>Client</th>
+                  <th>Language</th>
+                  <th>Case Number</th>
+                  <th>Due Date</th>
+                  <th>Select</th>
+                </tr>
+              </thead>
+              <tbody>
+                {loading ? (
+                  <div>loading</div>
+                ) : (
+                  cases.map((task) => (
+                    <tr key={`${task.id} task`}>
+                      <td>
+                        {task.first_name} {task.last_name}
+                      </td>
+                      <td>
+                        <span className="uk-label">
+                          {lang_short[task.fromLanguage]} &#9658;{" "}
+                          {lang_short[task.toLanguage]}
+                        </span>
+                      </td>
+                      <td>{task.case_number}</td>
+                      <td>{formatDate(task.due_date)}</td>
+                      <td>
+                        <input
+                          type="checkbox"
+                          value={task.checked}
+                          onClick={() => {
+                            let tempCases = cases;
+                            Object.values(tempCases).forEach((value) => {
+                              if (value.id === task.id) {
+                                value.checked = !task.checked;
+                                value.translator.first_name = first_name;
+                                value.translator.last_name = last_name;
+                                value.translator.id = translator_id;
+                              }
+                            });
+                            setCases(tempCases);
+                          }}
+                        />
+                      </td>
+                    </tr>
+                  ))
+                )}
+              </tbody>
+            </table>
+          </div>
+          <div className="uk-modal-footer uk-text-right">
+            <button
+              className="uk-button uk-button-primary uk-modal-close"
+              type="button"
+              onClick={() => {
+                Object.values(cases).forEach((element) => {
+                  if (element.checked) {
+                    CaseService.updateStatus(element.id, "Assigned");
+                    CaseService.updateTranslator(element.id, first_name, last_name, 0);
+                  }
```

Comment:
with suggested change below, it will be

`CaseService.assignCase(element.id, "Assigned", {first_name, last_name, translator_id});`

don't forget to test if it works, please

Reply:
i fixed